### PR TITLE
Add type hints to test_backend_test.py

### PR DESCRIPTION
**Pull request by smessmer**

### CODE REVIEW

**Comment by smessmer** [comment]
@onnxbot retest this please

File: onnx/test/test_backend_test.py [hunk 1]
```diff
@@ -49,7 +62,7 @@ def supports_device(cls, device):
         return False
 
 
-backend_test = onnx.backend.test.BackendTest(DummyBackend, __name__)
+backend_test = onnx.backend.test.BackendTest(DummyBackend(), __name__)
```
Comment (bddppq) [inline onnx/test/test_backend_test.py]: Backend is intended to be used as singleton

**Comment by smessmer** [comment]
@onnxbot retest this please

**Comment by smessmer** [comment]
@onnxbot retest this please

**Comment by smessmer** [comment]
The CI here was still failing, but (probably) unrelated. Let's see if this breaks CI on master.

**Pull request merged**